settings: Prevent long email spills over in notice.

palashb01 opened the pull request:
- Improves the readability of alert notification banners by breaking long words into multiple lines and wrapping the content to fit the viewport.
- Prevent the notification banner content from spilling over by wrapping it inside the notice.
- Decrease the gap between the account heading and the 'account-alert-notification' banner. 

Fixes: #23653

------------------

**Screenshots and screen captures:**
<details>
<summary>Before:</summary>

<br>

![Screenshot from 2023-02-14 19-18-50](https://user-images.githubusercontent.com/66828942/218757160-7d14a0c3-ce77-42b1-a20e-95718b4e0e21.png)

</details>

**New Changes:**

<details>
<summary>When the  email is short:</summary>

<br>

![Screenshot from 2023-03-05 12-31-05](https://user-images.githubusercontent.com/66828942/222946758-3774bf1f-c8ff-406e-8191-6a0259b0abe8.png)


</details>

<details>
<summary>When the  email is long:</summary>

<br>


![Screenshot from 2023-03-10 00-20-15](https://user-images.githubusercontent.com/66828942/224127536-6e0a69d9-435a-4cf4-809b-e5bf09f0ef09.png)


</details>
<details>
<summary>View port changes:</summary>

<br>

![view_port_changes](https://user-images.githubusercontent.com/66828942/224127602-4b1306b6-e0fb-4263-b986-fa2a958e08dd.gif)

</details>

<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [x] Explains differences from previous plans (e.g., issue description).
- [x] Highlights technical choices and bugs encountered.
- [x] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [x] Responsiveness and internationalization.
- [x] Strings and tooltips.
- [x] End-to-end functionality of buttons, interactions and flows.
- [x] Corner cases, error conditions, and easily imagined bugs.
</details>

palashb01 commented:
Hey @alya @sahil839, could you please review the PR and suggest any necessary changes? Thank you.

sahil839 commented on static/styles/settings.css:
```diff
@@ -501,16 +501,22 @@ input[type="checkbox"] {
     width: auto !important;
+    white-space: break-spaces !important;
 
+    word-break: break-word;
```
`break-word` is depreceated as mentioned in https://developer.mozilla.org/en-US/docs/Web/CSS/word-break#values.

palashb01 replied:
Have changed it to `break-word: normal` with `overflow-wrap: anywhere`

sahil839 reviewed (commented):
Posted one comment. Also, I feel the space between the `Account` heading and the box can be decreased.

sahil839 commented:
Also, this change shifts the notification box closer to the heading in other sections too as you removed `margin-left: 10px` property.

palashb01 commented:
> Also, this change shifts the notification box closer to the heading in other sections too as you removed `margin-left: 10px` property.

Hey @sahil839 ,I have reverted the `margin-left` to 5px and removed the margin-right from the heading. Could you please review the PR, Thanks.

sahil839 commented on web/styles/settings.css:
```diff
@@ -513,11 +513,15 @@ input[type="checkbox"] {
     margin-top: 14px;
-    margin-left: 10px;
+    margin-bottom: 5px;
+    margin-left: 5px;
```
These margin changes changes the UI for other notices too. I think we should only do CSS changes for the notice in that particular panel.

![Screenshot from 2023-02-27 22-55-09](https://user-images.githubusercontent.com/35494118/221636762-ae5ce355-af37-4e1f-b5e8-5ba7d9c416fd.png)

palashb01 replied:
have created a new classname `account-alert-notification` and made the new margin changes for the account alert notice only.

sahil839 commented on web/styles/settings.css:
```diff
@@ -513,11 +513,15 @@ input[type="checkbox"] {
     width: auto !important;
+    white-space: break-spaces !important;
 
+    word-break: normal;
```
I think we can remove this since default for `word-break` is `normal`.

palashb01 replied:
made the change.

sahil839 reviewed (commented):
Posted a couple of comments. Also, I think the gap between the heading and notice can be reduced.
![Screenshot from 2023-02-27 22-52-21](https://user-images.githubusercontent.com/35494118/221644304-f32f53c3-1c73-42e0-a725-c06b31902718.png)

palashb01 commented:
> Posted a couple of comments. Also, I think the gap between the heading and notice can be reduced. ![Screenshot from 2023-02-27 22-52-21](https://user-images.githubusercontent.com/35494118/221644304-f32f53c3-1c73-42e0-a725-c06b31902718.png)

reduced the `margin-top` to 10px, which is the default margin for all notice banners. @sahil839, I have made all the suggested changes. Could you please review the PR, Thanks.

sahil839 commented on web/styles/settings.css:
```diff
@@ -513,14 +513,20 @@ input[type="checkbox"] {
     background-color: transparent;
     border-radius: 4px;
-    margin-top: 14px;
-    margin-left: 10px;
```
I think this has been accidentally removed.

palashb01 replied:
Reverted.

sahil839 commented on web/styles/settings.css:
```diff
@@ -513,14 +513,20 @@ input[type="checkbox"] {
     border-radius: 4px;
-    margin-top: 14px;
-    margin-left: 10px;
+    margin-top: 10px;
```
If we want this change this should be added only to the notice in accounts section. And if we want this change for all other notices as well, this should be a separate commit.

palashb01 replied:
I have made these changes into separate class `account-alert-notification` of accounts section.

sahil839 reviewed (commented):
Posted a couple of comments

And I think there is still a lot of space between the `Account` heading and the notice for very long emails.
I think one solution can to remove the `margin-top` for specifically this notice and instead align it using `vertical-align`. And then we can probably in separate commit try to remove hard-coded margin for other notices too with proper testing.

palashb01 commented:
Hey @sahil839 , I have made all the changes suggested by you, could you please review the PR, thanks.

sahil839 commented:
The alert still looks odd when it is shifted below due to the left side margin. To handle both the cases for long and short messages in alert, we should instead a right margin to heading and make the left margin of alert as 0 for this particular section only for now.

And also would be better to have same space both above and below the notice in case of long messages inside it.

sahil839 commented:
Also, the first commit message incorrectly mentions things that are not changed.

palashb01 commented:
> Also, the first commit message incorrectly mentions things that are not changed.

 I'm not sure why the change wasn't pushed earlier, but I have made the necessary changes now. I set the `word-break` property to `break-all` because if the email is long and does not fit in one line, it wraps to another line, which may not look good in certain conditions. This change has been made for all the notices because a similar issue in other notices could also be fixed with this change.

- Without word-break:

![Screenshot from 2023-03-06 23-42-38](https://user-images.githubusercontent.com/66828942/223207730-1fcd95e9-dc68-4701-86b7-5fbd1dcbd5f8.png)

- With word-break:

![Screenshot from 2023-03-06 23-42-43](https://user-images.githubusercontent.com/66828942/223207820-64c5eb64-7b64-4fb4-abbe-4c15b1c413c0.png)

palashb01 commented:
> Also, the first commit message incorrectly mentions things that are not changed.

I initially included a `word-break` property to reduce the height of the banner in case of a long email name. However, after the recent conversation on CZO [Thread](https://chat.zulip.org/#narrow/stream/101-design/topic/alert_notice.3A.20Prevent.20long.20email.20spills.20.2323653) I have reverted that.

@sahil839 I have made all the suggested changes, could you please review the PR again, Thanks.

sahil839 commented on web/styles/settings.css:
```diff
@@ -513,14 +517,23 @@ input[type="checkbox"] {
+    &.account-alert-notification {
+        margin-bottom: 10px;
+        margin-left: 0;
+        margin-top: 0;
```
Can we change this to use single `margin` property?

palashb01 replied:
Changed it to a single margin statement and updated the commit messages according to your suggestion.

sahil839 reviewed (commented):
Posted one comment. Also, I think `settings` would be a better prefix for commit messages and the commit summary for second commit is too long I guess, would be good to make it short as per the guidelines.

sahil839 commented:
The position banner in cases when email is short is not exactly same as for `Saving` banner for other settings. But still looks fine and would be hard to write CSS for both the short and long banners perfectly. @alya I think you can test this now once.

timabbott commented:
Merged, sorry for the delay -- looks like this never got the "integration review" tag added.
